### PR TITLE
 [Playlist] feat: add dynamic group noun through config

### PR DESCRIPTION
**Pull request by antoniobergas**
## Hey, I just made a Pull Request!

I added some config properties for the Playlist plugin to be able to change dynamically the group noun in all UI components

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by kuangp — CHANGES_REQUESTED** [review]
cc @benjdlambert if you have some thoughts about this change as it relates to the [RFC](https://github.com/backstage/backstage/issues/16876) you recently opened, specifically regarding doing simple UI text customization in plugins

**Comment by antoniobergas** [comment]
thanks for the feedback @kuangp I will make the changes as soon as I have some time.

Can you maybe guide me with the test CI issue?
image

I check locally but don't know why that it's failing

File: plugins/playlist/config.d.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+export interface Config {
+  playlist: {
+    /**
+     * (Optional) The plural noun for the entities grouping that will shown in the UI; leave empty for `PLaylists`.
+     * @visibility frontend
+     */
+    groupPluralNoun: string;
```
[the remark this answers is not in the document]
Comment (antoniobergas) [reply plugins/playlist/config.d.ts:23]: Yeah, having pluralize it will make it easier and title make more sense 👍🏻

**Comment by antoniobergas** [comment]
I found out that the issue in the test its related to use the new useTitle hook at the same time that the usePlaylistList hook.
image
image

image


The tests of usePlaylistList hook its also declaring a configApiRef. I have not that much expertise in React to know why this it's producing an error in the test neither how to fix it. 

If someone can guide me I will appreciate it 😊

File: plugins/playlist/src/hooks/useTitle.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { configApiRef, useApi } from '@backstage/core-plugin-api';
+import pluralize from 'pluralize';
+
+export function useTitle(inPlural: boolean, inLowerCase: boolean) {
```
[the remark this answers is not in the document]
Comment (antoniobergas) [reply plugins/playlist/src/hooks/useTitle.ts:20]: Make sense! I will make the changes

**Review by kuangp — APPROVED** [review]
Looks good to me! Thanks for the great contribution! 

over to @benjdlambert for maintainers review

**Comment by antoniobergas** [comment]
Done, I think I am not forgetting anything.

Thanks a lot for the help and the guidance @kuangp 😊

**Comment by benjdlambert** [comment]
Sorry - just coming round to this now.

Is there a reason for the preference of `config` over a prop to pass in the `plugin` itself? It would just be unfortunate not to follow the current (most probably going to change with #16876) patterns that we have in place for this.

You're right @kuangp that it deffo fits into the i18n category of that RFC, would like to try and follow the patterns that we have to day, even it they are changing :pray:

Thoughts?

**Comment by kuangp** [comment]
hmm my recommendation of using a config here is same reasons laid out in the RFC regarding prop drilling to multiple various components to do a simple text customization. It's also inspired by the current `app.title` config that allows you to "rebrand" Backstage and in the same manner since this change is about rebranding the "Playlist" term (originally named to follow the Backstage/Spotify naming theme 😄), I thought it made sense to leverage the same mechanism ie. a company will probably rebrand "Backstage" and "Playlists" in a consistent manner to both follow whatever naming theme they wish to use.

**Comment by antoniobergas** [comment]
I am quite new in Backstage and don't know at all the impact of this.

If I understood correct, there's 3 flavors to pass custom data to plugins right?
- Vía configApi
- Vía props
- Vía i18n (just text data)(in progress)

In a first attepmt I also thought about the props but after kuangp mentioned the configApi, It looks quite more clean solution (and also easy to use from a user perspective)

Which are the pros and cons of each flavor ?

**Comment by antoniobergas** [comment]
Hi guys! Any suggestions of the next steps for this?

**Comment by kuangp** [comment]
@benjdlambert thoughts on my reasoning ?

**Comment by benjdlambert** [comment]
Apologies - had a busy few weeks and just getting back round to this now.

Yeah I see your reasoning @kuangp, it deffo fits into the internationalisation part I think, which I guess is going to come from config eventually. I don't really have a strong opinion to be honest, maybe let's just keep it in config for now then.

I guess we could also use this example as something to drive the i18n part of that RFC too if someone wants to have a stab at that after? :pray:

**Review by benjdlambert — APPROVED** [review]

**Comment by antoniobergas** [comment]
Thanks @benjdlambert ! I will try to keep an eye to the RFC if we need to adjust it later.

Thanks also @kuangp for guide me 😊

**Pull request merged**